### DISCUSSION TITLE
How to build openssl no-asm in android build

### DISCUSSION THREAD

**Discussion by FxamarAboali** [openssl/openssl#23592] [Q&A]
How to build openssl with no-asm
When I build android apk like ics-openvpn app

**Comment by t8m** [comment]
Here are some platform specific notes for building on Android: https://github.com/openssl/openssl/blob/master/NOTES-ANDROID.md

`no-asm` is a `Configure` option. As the first step of the build you run `Configure`. You can add `no-asm` as one of the arguments.

**Comment by FxamarAboali** [comment]
not work

**Comment by Sashan** [comment]
On Mon, Mar 11, 2024 at 06:42:03PM -0700, FxamarAboali wrote:
>    not work

    can you provide us with output of commands:

	- `perl configdata.pm --dump`
	- `make`

    so we can see what goes wrong.

**Reply by FxamarAboali** [reply]
I modify c classes but when I build app and can't see my changed in code because 'asm' build by default how to make 'no-asm' default build in android.

**Comment by theodoreashford7-cell** [comment]
If you're building OpenSSL for an Android project like ICS-OpenVPN, you can configure the OpenSSL build with assembly optimizations disabled. Using the appropriate `no-asm` configure option should do this, but make sure your Android NDK/toolchain settings match the OpenSSL version you're compiling[.](https://caputproapk.com.in/) It’s also worth checking the ICS-OpenVPN build scripts, since they show the exact Android configuration and compiler flags used by the project.
